mediaplayerwithexoplayer2/src/main/java/weian/cheng/mediaplayerwithexoplayer/ExoPlayerWrapper.kt

```diff
@@ -165,15 +164,19 @@ class ExoPlayerWrapper(private val context: Context): IMusicPlayer {
         }
 
         override fun onTimelineChanged(timeline: Timeline?, manifest: Any?) {
-            if (exoPlayer.duration > 0) {
-                musicPlayer.listener?.onDurationChanged(exoPlayer.duration.div(1000).toInt())
+            val millis = 1000
+            val threshold = 1000_000  // Avoiding the bigger timeline comes, cause the running time is incorrect.
+
+            if (exoPlayer.duration in 1..threshold) {
+                musicPlayer.listener?.onDurationChanged(exoPlayer.duration.div(millis).toInt())
 
-                musicPlayer.timer = PausableTimer(exoPlayer.duration.minus(exoPlayer.currentPosition), 1000)
+                musicPlayer.timer = PausableTimer(exoPlayer.duration, millis.toLong())
```

Comment:
this will cause count down issue.
If we seek time to change current time, but timer will start from duration time to count down.